modules/user/user_test.go

```diff
@@ -2,21 +2,40 @@ package user
 
 import (
 	"os"
+	"os/exec"
+	"runtime"
+	"strings"
 	"testing"
 )
 
-func TestCurrentUsername(t *testing.T) {
-	os.Setenv("USER", "")
-	os.Setenv("USERNAME", "foobar")
+func getWhoamiOutput() (string, error) {
+	output, err := exec.Command("whoami").Output()
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(string(output[:])), nil
+}
 
+func TestCurrentUsername(t *testing.T) {
 	user := CurrentUsername()
-	if user != "foobar" {
-		t.Errorf("expected foobar as user, got: %s", user)
+	if len(user) <= 0 {
+		t.Errorf("expected non-empty user, got: %s", user)
 	}
-
-	os.Setenv("USER", "gitea")
+	// Windows whoami is weird, so just skip remaining tests
+	if runtime.GOOS == "windows" {
+		return
```

Comment:
Please use t.Skip in place of return with some explaination like the comment two line before.